### DISCUSSION TITLE
Copy content between 2 different self hosted S3 Storages

### DISCUSSION THREAD

**Discussion by Dannhausen** [iterate-ch/cyberduck#15746] [Q&A] [answered]
Hi,
i´m trying to copy content between two different S3 storages which are self hosted and have different credentials. 

I tried it the following way, but always get  errors for the destination bucket

```
duck --username "xxx" --password "xxx" --copy "s3://IP-Source/Path" "s3://IP-Destination/Path" --username "yyy" --password "yyy" --existing compare
```

i can do this with little change over a lokal temporary copy. but i would like to do it directly for some reasons ;)

i hope someone has an answer for me :)

**Comment by dkocher** [comment, marked as answer]
Unfortuantely the [documentation](https://docs.duck.sh/cli/) is lacking [^1]. The parameters `username` and `password`  cannot be set twice. Instead try to include the username in the URL like `s3://accesskey:secretkey@ip/path`. I have opened [^2] to clarify documentation. 

[^1]: https://github.com/iterate-ch/docs/issues/72
[^2]: iterate-ch/docs#491

**Reply by Dannhausen** [reply]
Thx for the fast answer. This helped a lot :)
